draft-ietf-privacypass-architecture.md

```diff
@@ -544,6 +545,76 @@ and guidelines on how to deploy the protocol to minimize any privacy impacts.
 Any extension to the Privacy Pass protocol MUST adhere to the guidelines
 specified in {{issuer-role}} for managing Issuer public key data.
 
+## Information Flow {#flow}
+
+The end-to-end run of redemption and issuance protocol involves information
+flowing between Issuer, Origin, Client, and Attester. That information can
+have implications on the privacy goals that Privacy Pass aims to provide
+as outlined in {{privacy-and-trust}}. In this section, we describe the flow
+of information between each party. How this information affects the privacy
+goals in particular deployment models is further discussed in {{deployment}}.
+
+### Token Challenge Flow {#challenge-flow}
+
+To use Privacy Pass, Origins choose an Issuer from which they are willing to
+accept tokens. Origins then construct a token challenge using this specified
+Issuer and the redemption context it shares with the Client. This token
+challenge is then delivered to a Client. The token challenge conveys
+information about the Issuer and the redemption context, such as whether the
+Origin desires a per-Origin or cross-Origin token.
+
+### Attestation and Issuance Flow {#issue-flow}
+
+Clients use the issuance protocol to produce a token bound to a token
+challenge. In doing so, there are several ways in which the issuance protocol
+contributes information to the attestation context and issuance context.
+For example, a token request may contribute information to the attestation
+or issuance contexts as described below.
+
+- Issuance protocol. The type of issuance protocol can contribute information
+about the Issuer's capabilities to the attestation or issuance contexts.
```

Comment:
> Why would the issuance protocol choice leak information to the issuance context? That's not immediately obvious.

As an example, maybe an Issuer only supports the rate-limiting issuance protocol for one origin. This would be pretty silly, but it would be possible, and as a result, the issuance protocol immediately reveals unwanted information to the issuer.

> I think that what you are saying is that the choice made by the Origin regarding the issuance protocol might be used to carry information from the Origin to the Issuer if the Client requests issuance when it receives a challenge. If the issuance protocol choice is exposed to the Attester, then it too learns this information. If there are multiple options available, some information about the Client can be encoded in the choice between those options.

That's a good clarification, yeah. I'll include it.